vms/evm/subscriber.go

```diff
@@ -161,58 +162,78 @@ func (s *subscriber) ProcessFromHeight(height *big.Int) error {
 		return err
 	}
 
-	// Only process logs if the provided height is not nil. Otherwise, simply update the database with
-	// the latest block height
-	if height != nil {
-		// Cap the number of blocks to process to MaxBlocksToProcess
-		toBlock := big.NewInt(0).SetUint64(latestBlock)
-		if height.Cmp(big.NewInt(0).Add(toBlock, big.NewInt(-MaxBlocksToProcess))) < 0 {
-			s.logger.Warn(
-				fmt.Sprintf("Requested to process too many blocks. Processing only the most recent %s blocks", MaxBlocksToProcess),
-				zap.String("requestedBlockHeight", height.String()),
-				zap.String("latestBlockHeight", toBlock.String()),
-			)
-			height = big.NewInt(0).Add(toBlock, big.NewInt(-MaxBlocksToProcess))
-		}
+	// Cap the number of blocks to process to MaxBlocksToProcess
+	toBlock := big.NewInt(0).SetUint64(latestBlock)
+	if height.Cmp(big.NewInt(0).Add(toBlock, big.NewInt(-MaxBlocksToProcess))) < 0 {
```

Comment:
sorry missed this in previous reviews, but I'm thinking whether we should error out if there are too many blocks to process. Or, maybe we should let the max amount be configured by the user, in case they do want to just process all of the previous blocks.

Also, what happens if the subscriber sees new logs before the relayer creation finishes? I'm worried that there would be nil reference since the relayer hasnt been set yet and we try to call `relayer.RelayMessage()`

Reply:
I'm going to leave the max blocks behavior as is for now, as I think we should step back and discuss how to handle the case in which the relayer is severely behind. We should the tradeoff between reasonable look back limits and interaction among concurrently running relayer instances. I also think the scope of this PR is starting to balloon, and at the moment we're not very unlikely to bump against these limits anyway.

Reply:
> Also, what happens if the subscriber sees new logs before the relayer creation finishes? I'm worried that there would be nil reference since the relayer hasnt been set yet and we try to call relayer.RelayMessage()

This won't be an issue since we don't call `RelayMessage` until after `NewRelayer` returns anyway. If a new message is seen before relayer creation finishes, it'll just be queued in the subscriber channel until it is read [here].(https://github.com/ava-labs/awm-relayer/blob/ba3f2fb6c794b05c33c9a46b83966fa03faa12ec/main/main.go#L233)

Reply:
sounds good, ye we can make that separate case, can you create an issue to keep track?

Ahh ye covers the case I was worried about.